rcl/include/rcl/event.h

```diff
@@ -30,13 +30,15 @@ extern "C"
 typedef enum rcl_publisher_event_type_t
 {
   RCL_PUBLISHER_OFFERED_DEADLINE_MISSED,
-  RCL_PUBLISHER_LIVELINESS_LOST
+  RCL_PUBLISHER_LIVELINESS_LOST,
+  RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS
```

Comment:
So unlike rmw (https://github.com/ros2/rmw/pull/193/files#r370904671), I think `RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS` makes total sense. Because the publisher is doing something. In the rmw case we're describing an event, so the order makes more sense in the other order.